web/client-api/src/main/java/io/deephaven/web/client/api/JsPartitionedTable.java

```diff
@@ -81,20 +82,10 @@ public Promise<JsPartitionedTable> refetch() {
 
             return w.getExportedObjects()[0].fetch();
         }).then(result -> connection.newState((c, state, metadata) -> {
-            JsTable keyTable = (JsTable) result;
-            SelectOrUpdateRequest view = new SelectOrUpdateRequest();
-            view.setSourceId(keyTable.state().getHandle().makeTableReference());
-            view.setResultId(state.getHandle().makeTicket());
-            view.setColumnSpecsList(descriptor.getKeyColumnNamesList());
-            connection.tableServiceClient().view(view, metadata, (fail, success) -> {
-                keyTable.close();
-                c.apply(fail, success);
-            });
-        }, "view only keys columns")
+            baseTable = (JsTable) result;
+        }, "get base table")
```

Comment:
Removing the contents of the lambda here is causing the test (or any PartitionedTable creation) to hang - the `c` param must be called with success or failure, not unlike how a Promise constructor call must have either `resolve` or `reject` invoked. Failing to do that is causing this to hang forever.

We can remove the newState (we dont need a newState, result is already a Table), the refetch (since there is nothing to fetch), and the .then() that returns a new JsTable, and just continue with the result table.

That doesn't fix the tests, new failure... will catch up later on slack.

Reply:
Doh, missed adding this line to the next .then(), before keyColumnTypes

```
baseTable = (JsTable) result;
```

Then tests pass.

Reply:
Repushed after verifying 😄 .